packages/npm/send/android/src/main/java/com/candlefinance/send/SendModule.kt

```diff
@@ -1,25 +1,173 @@
 package com.candlefinance.send
 
+import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
 import com.facebook.react.bridge.ReactMethod
-import com.facebook.react.bridge.Promise
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.internal.http.HttpMethod
+import okio.Buffer
+import okio.ByteString.Companion.decodeBase64
+import okio.GzipSource
+import okio.IOException
+import java.util.concurrent.TimeUnit
+
+// FIXME: Actually check charset parameter first if set, then fall back to MIME type
+private fun bodyIsUTF8(contentTypeHeader: String?, utf8ContentTypes: List<String>): Boolean {
+  return contentTypeHeader != null && utf8ContentTypes.contains(contentTypeHeader.split(';')[0])
```

Comment:
Not sure if this is actually better, but at least it's shorter.....up to you which to use, just making the possibility known
```suggestion
  return utf8ContentTypes.contains(contentTypeHeader?.split(';')?.get(0))
```